backend/apis/leads/update-lead.api.js

```diff
@@ -83,6 +83,47 @@ app.patch("/api/leads/:id", (req, res, next) => {
       );
     }
 
+    const leadServicesChanged = atLeastOne(req.body, "leadServices");
+
+    if (leadServicesChanged && fullLead.leadServices.length > 0) {
+      queries.push("UPDATE leads SET modifiedBy = ? WHERE id = ?;");
+      queryData.push(userId, leadId);
+
+      const oldLeadServices = [...oldLead.leadServices];
+      const oldLeadServiceIds = oldLeadServices.map(service => {
+        return service.id;
+      });
+      const fullLeadServiceIds = [...fullLead.leadServices];
+
+      const newLeadServiceIds = [];
+
+      for (let i = 0; i < fullLeadServiceIds.length; i++) {
```

Comment:
The way this API is implemented does not allow for interest in a service to ever be deleted. I think that absence from the leadServices array should mean that there is no interest in the service.

Also, your algorithm here with the nested for loops could be simplified by using lodash [difference()](https://lodash.com/docs/4.17.15#difference):

```js
const { difference, flatten } = require('lodash')

// do a query to get all the possible services. Maybe do this up top at the same time you're getting the old lead
pool.query(`SELECT id FROM services`, (err, result) => {
  const allServiceIds = result[0]
  const servicesToDelete = difference(fullLead.leadServices, allServiceIds)
  queries.push(...servicesToInsert.map(s => `DELETE FROM leadServices WHERE leadId = ? AND serviceId = ?`))
  queryData.push(...flatten(servicesToInsert.map(s => ([leadId, s.id]))))

  const servicesToInsert = difference(oldLead.services.map(s => s.id), fullLead.leadServices)
  queries.push(...servicesToInsert.map(s => `INSERT INTO leadServices (leadId, serviceId) VALUES (?, ?);`))
  queryData.push(...flatten(servicesToInsert.map(s => ([leadId, s.id]))))
})

```